internal/speedtest/librespeed.go

```
@@ -119,17 +121,26 @@ func (r *LibrespeedRunner) RunTest(ctx context.Context, opts *types.TestOptions)
-		return nil, fmt.Errorf("librespeed-cli failed: %v: %s", err, string(output))
+	// Capture stdout and stderr separately: librespeed-cli writes the JSON
+	// report to stdout, while progress/UI lines and telemetry errors (emitted
+	// when --share is used) go to stderr. Keeping them apart avoids stderr
+	// noise corrupting the JSON we parse.
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		log.Error().Err(err).Str("stdout", stdout.String()).Str("stderr", stderr.String()).Msg("librespeed-cli failed")
+		return nil, fmt.Errorf("librespeed-cli failed: %v: %s", err, strings.TrimSpace(stderr.String()))
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)